extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -39,12 +42,12 @@ export class DeployDatabaseDialog {
 		this.initializeDialog();
 		this.dialog.okButton.label = constants.deployDialogOkButtonText;
 		this.dialog.okButton.enabled = false;
-		this.dialog.okButton.onClick(async () => await this.deploy());
+		this.dialog.okButton.onClick(async () => await this.deployClick());
 
 		this.dialog.cancelButton.label = constants.cancelButtonText;
 
 		let generateScriptButton: azdata.window.Button = azdata.window.createButton(constants.generateScriptButtonText);
-		generateScriptButton.onClick(async () => await this.generateScript());
```

Comment:
the return type of the onClick is disposable, please handle the disposables properly.